src/producer/mod.rs

```diff
@@ -129,6 +134,25 @@ impl<S: Read + Write> Producer<S> {
             .issue(&QueueControl::new(QueueAction::Resume, queues))?
             .await_ok()
     }
+
+    /// Initiate a new batch of jobs.
+    #[cfg(feature = "ent")]
+    pub fn start_batch(&mut self, batch: Batch) -> Result<BatchHandle<'_, S>, Error> {
+        let bid = self.c.issue(&batch)?.read_bid()?;
+        Ok(BatchHandle::new(bid, self))
+    }
+
+    /// Open an already existing batch of jobs.
+    #[cfg(feature = "ent")]
+    pub fn open_batch(&mut self, bid: String) -> Result<BatchHandle<'_, S>, Error> {
```

Comment:
Can we make this return `Result<Option` so that if there is no batch with the given `bid` we return `Ok(None`?

Reply:
I just realized I never tested this case. Checked it: trying to open a never-actually-created batch with id, say, `non-existent-batch-id` will return an error `Protocol(Internal { msg: "No such batch non-existent-batch-id" })`. Will  look into it..

Reply:
Add `maybe_bid` method on the `ReadToken` to handle this case